osquery/database/tests/database_tests.cpp

```diff
@@ -114,4 +115,75 @@ TEST_F(DatabaseTests, test_ptree_upgraded_to_rj) {
   auto ulepoch = std::stoull(query_epoch);
   EXPECT_EQ(ulepoch, 1521583712U);
 }
+
+static void ptree_to_json(const pt::ptree& pt_in, const JSON& rj_in) {
+  /* Convert from ptree to string */
+  std::stringstream ss;
+  pt::write_json(ss, pt_in);
+
+  /* Convert from ptree string to rj string */
+  std::string conversion_out;
+  EXPECT_TRUE(ptreeToRapidJSON(ss.str(), conversion_out));
+
+  /* Convert rj string back to rj */
+  JSON rj_out;
+  EXPECT_TRUE(rj_out.fromString(conversion_out));
+
+  /* Compare original rj with output */
+  EXPECT_EQ(rj_in.doc(), rj_out.doc());
+}
+
+TEST_F(DatabaseTests, test_ptree_to_rj_empty) {
+  /* '{}' */
+
+  pt::ptree pt_in;
+
+  JSON rj_in;
+
+  ptree_to_json(pt_in, rj_in);
+}
+
+TEST_F(DatabaseTests, test_ptree_to_rj_object) {
+  /* '{ "k" : "v" }' */
+
+  pt::ptree pt_in;
+  pt_in.put("k", "v");
+
+  JSON rj_in = JSON::newObject();
+  rj_in.add("k", "v");
+
+  ptree_to_json(pt_in, rj_in);
+}
+
+TEST_F(DatabaseTests, test_ptree_to_rj_array) {
+  /* '{ "" : "a", "" : "b" }' */
+
+  pt::ptree pt_in;
+  pt_in.put("", "a");
```

Comment:
I don't think this achieve the above (incorrect) JSON through property trees.

I *think* the way this is done:
```cpp
pt_in.push_back(std::make_pair("", "a"));
```